### PR TITLE
Fix `get_model_info` to provide logged model info

### LINKED ISSUE #16706: [BUG] `mlflow.models.get_model_info()` does not provide `LoggedModel`'s information
### Issues Policy acknowledgement

- [x] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### MLflow version

- Client: 3.1.1
- Tracking server: 3.1.1

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 24.04
- **Python version**: 3.12

### Describe the problem

`mlflow.models.get_model_info()` does not provide `model_id`, `metrics`, `params`, `tags`, `creation_timestamp`, and `name`.

`get_model_info()` is invoked without passing the `logged_model` argument. As `_logged_model` is not provided, the fields are empty.
https://github.com/mlflow/mlflow/blob/72a076664fe5f04acbccf1819c42c84be64cbcf6/mlflow/models/model.py#L1449

https://github.com/mlflow/mlflow/blob/72a076664fe5f04acbccf1819c42c84be64cbcf6/mlflow/models/model.py#L340-L390

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```python
import mlflow

mlflow.set_tracking_uri("http://localhost:5000")

model = mlflow.create_external_model(
    model_type="agent",
)

mv = mlflow.register_model(model.model_uri, "example_agent")

model_info = mlflow.models.get_model_info("models:/example_agent/1")

print(model_info.__dict__)

```

### Stack trace

```
REPLACE_ME
```

### Other info / logs

```
Downloading artifacts: 100%|███████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 1597.83it/s]
{'_artifact_path': 'mlflow-artifacts:/0/models/m-22da4372074a4fa0b2978d4126cc33e5/artifacts', '_flavors': {}, '_model_uri': 'models:/m-22da4372074a4fa0b2978d4126cc33e5', '_model_uuid': 'm-22da4372074a4fa0b2978d4126cc33e5', '_run_id': '', '_saved_input_example_info': None, '_signature': inputs: 
  [boolean (required)]
outputs: 
  [boolean (required)]
params: 
  None
, '_utc_time_created': '2025-07-12 14:56:17.668171', '_mlflow_version': '3.1.1', '_metadata': {'mlflow.model.isExternal': True}, '_prompts': None, '_registered_model_version': None, '_env_vars': None, '_logged_model': None}
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/prompt`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/16713?quickstart=1)

#### Install mlflow from this PR

```
# mlflow
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/16713/merge
# mlflow-skinny
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/16713/merge#subdirectory=skinny
```

For Databricks, use the following command:

```
%sh curl -LsSf https://raw.githubusercontent.com/mlflow/mlflow/HEAD/dev/install-skinny.sh | sh -s pull/16713/merge
```

</p>
</details>

### Related Issues/PRs

Fix #16706

### What changes are proposed in this pull request?

Minimum example to reproduce the error:

```python
import mlflow


def model(model_input: list[str]) -> list[str]:
    return model_input


model_info = mlflow.pyfunc.log_model(
    name="test_model", python_model=model, input_example=["a", "b", "c"]
)
model_info = mlflow.models.get_model_info(model_info.model_uri)
assert model_info.model_id == model_info.model_id
```

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

Fix `get_model_info` to include logged model info.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/prompt`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/models/model.py [hunk 1]
```diff
@@ -344,7 +344,7 @@ def model_id(self) -> str:
 
         :getter: Gets the model ID of the logged model
         """
-        return self._logged_model.model_id
+        return self._logged_model and self._logged_model.model_id
```
Comment (harupy) [inline mlflow/models/model.py]: ` self._logged_model.model_id` throws an `AttributeError` if ` self._logged_model` is `None`.

Comment (serena-ruan) [reply mlflow/models/model.py]: I would prefer `self._logged_model.model_id if self._logged_model else None`, this `and` operation is a bit implicit

File: mlflow/models/model.py [hunk 2]
```diff
@@ -678,6 +678,9 @@ def get_model_info(self, logged_model: Optional[LoggedModel] = None) -> ModelInf
         Create a :py:class:`ModelInfo <mlflow.models.model.ModelInfo>` instance that contains the
         model metadata.
         """
+        if logged_model is None:
+            client = mlflow.MlflowClient()
+            logged_model = client.get_logged_model(model_id=self.model_id)
```
Comment (serena-ruan) [inline mlflow/models/model.py]: Can we use `mlflow.get_logged_model`?

File: tests/models/test_model.py [hunk 3]
```diff
@@ -713,3 +713,14 @@ def predict_fn(model_input: list[str]):
...
+
+
+def test_get_model_info_with_logged_model():
+    def model(model_input: list[str]) -> list[str]:
+        return model_input
+
+    model_info = mlflow.pyfunc.log_model(
+        name="test_model", python_model=model, input_example=["a", "b", "c"]
+    )
+    model_info = mlflow.models.get_model_info(model_info.model_uri)
```
Comment (serena-ruan) [inline tests/models/test_model.py]:
```suggestion
    model_info2 = mlflow.models.get_model_info(model_info.model_uri)
```

File: tests/models/test_model.py [hunk 4]
```diff
@@ -713,3 +713,14 @@ def predict_fn(model_input: list[str]):
...
+    assert model_info.model_id == model_info.model_id
```
Comment (serena-ruan) [inline tests/models/test_model.py]: Can we assert model_info == model_info2

Comment (harupy) [reply tests/models/test_model.py]: `==` doesn't seem to work. I think `__eq__` is missing.

File: mlflow/models/model.py (hunk 1 above)
Comment (harupy) [reply mlflow/models/model.py]: sounds good!

File: mlflow/models/model.py (hunk 2 above)
Comment (harupy) [reply mlflow/models/model.py]: we can.

File: mlflow/models/model.py [hunk 5]
```diff
@@ -344,7 +344,7 @@ def model_id(self) -> str:
 
         :getter: Gets the model ID of the logged model
         """
-        return self._logged_model.model_id
+        return self._logged_model.model_id if self._logged_model else None
```
Comment (TomeHirata) [inline mlflow/models/model.py:347]: nit: shall we add a test case for `_logged_model = None`?

**Review by TomeHirata — APPROVED** [review]
LGTM

File: mlflow/models/model.py (hunk 5 above)
Comment (harupy) [reply mlflow/models/model.py:347]: `_logged_model = None` is just for safety. It's should be populated.

Comment (TomeHirata) [reply mlflow/models/model.py:347]: Is it populated for models logged with MLflow 2x?
